python/tests/conftest.py

```diff
@@ -253,9 +253,15 @@ class EncodingVersion(enum.IntEnum):
     V2 = 1
 
 
-@pytest.fixture(params=list(EncodingVersion))
-def encoding_version(request) -> EncodingVersion:
-    return request.param
+@pytest.fixture(scope="session")
+def only_test_encoding_version_v1():
```

Comment:
I don't really want this to become a general pattern. If we're going to alternate between encodings then that's fine, but I don't think we should be skipping v2 encoding for a whole test module

Reply:
I fully appreciate your point and definitely didn't intend for this to be widely used. Currently, this is only used by `test_filtering` which should not involve much encoding version handling.

My main motivation is the `test_filtering` "unit" test file has many Hypothesis tests and takes many minutes to complete. It might be better to move those to an integration test to keep the unit suite fast.